### ISSUE TITLE
Feature request: can be used in the browser

### ISSUE THREAD

**Issue by dobesv** [hayes/node-parquet#1]
I know it's called "node-parquet" but it would be great if this could run in the browser for consuming parquet report files.

**Comment by hayes** [comment]
Depending on your use case this might work.  So far my implementation makes
liberal use of BigInts.  While these are supported in most newer browsers,
polyfilling them can't be done efficiently from r older browsers.
Compression is another thing that gets complicated.  The compression libs
are all lazy loaded, and should be easy enough to exclude (since most of
them depend on native bindings).

It's too early for me to commit to specific use cases, but I think some
form of browser support in newer browsers with a limited subset of
compression options seems doable.

**Comment by dobesv** [comment]
Makes sense, just throwing it in there for consideration as the project evolves.

By the way have you looked at apache arrow?  They have a somewhat similar problem to solve and a JS library.  Might be able to share some code and have conversion/compatilibity between the two.
